code/game/objects/items/pinpointer.dm

```diff
@@ -8,105 +12,80 @@
 	throw_speed = 4
 	throw_range = 20
 	matter = list("metal" = 500)
-	var/obj/item/disk/nuclear/the_disk = null
-	var/active = 0
-
-
-	attack_self()
-		if(!active)
-			active = 1
-			workdisk()
-			to_chat(usr, "<span class='notice'>You activate the pinpointer</span>")
-		else
-			active = 0
-			icon_state = "pinoff"
-			to_chat(usr, "<span class='notice'>You deactivate the pinpointer</span>")
-
-	proc/workdisk()
-		if(!active) return
-		if(!the_disk)
-			the_disk = locate()
-			if(!the_disk)
-				icon_state = "pinonnull"
-				return
-		setDir(get_dir(src,the_disk))
-		switch(get_dist(src,the_disk))
-			if(0)
-				icon_state = "pinondirect"
-			if(1 to 8)
-				icon_state = "pinonclose"
-			if(9 to 16)
-				icon_state = "pinonmedium"
-			if(16 to INFINITY)
-				icon_state = "pinonfar"
-		spawn(5) .()
-
-	examine(mob/user)
-		..()
-		for(var/obj/machinery/nuclearbomb/bomb in GLOB.machines)
-			if(bomb.timing)
-				to_chat(user, "Extreme danger.  Arming signal detected.   Time remaining: [bomb.timeleft]")
+	var/active = FALSE
+	var/target = null
+
+/obj/item/pinpointer/proc/set_target(mob/user)
+	if (iscrashgamemode(SSticker.mode))
+		target = input("Select the item you wish to track.", "Pinpointer") as null|anything in GLOB.gamemode_key_items
+		return
+
+	var/obj/item/disk/nuclear/the_disk = locate()
+	if(the_disk)
+		target = the_disk
+
+/obj/item/pinpointer/attack_self(mob/user)
+	if(!active)
+		active = TRUE
+		set_target(user)
+		if(!target)
+			return
+		START_PROCESSING(SSobj, src)
+		to_chat(user, "<span class='notice'>You activate the pinpointer</span>")
+	else
+		active = FALSE
+		STOP_PROCESSING(SSobj, src)
+		icon_state = "pinoff"
+		to_chat(user, "<span class='notice'>You deactivate the pinpointer</span>")
+
+/obj/item/pinpointer/process()
+	if(!active)
```

Comment:
What will toggle the active state? Instead you should toggle processing manually rathern than this catch-all solution